### ISSUE TITLE
All observation of classes.

### ISSUE THREAD

**Issue by typesupply** [typesupply/defcon#34]
In many cases an observer needs to observe all glyphs in a font. That's only possible by subscribing to each glyph individually. That's cumbersome and becomes problematic when glyphs are added/removed. A better way to do this would be for an observer to observe all instances of a specific class. The observer would be notified for changes made to any object of that class.

This should be pretty easy to implement and shouldn't break anything. Here's an API sketch:

~~~ python
font.dispatcher.addObserver(
	observer=self,
	methodName="aGlyphChangedNotificationCallback",
	notification="Glyph.Changed",
	observable=font.glyphClass
)
~~~

Inside of the dispatcher there can be a loop like this:

~~~ python
for cls, observers in self._classObservers:
	if isinstance(objectSendingOutNotification, cls):
		for observer in observers:
			# filter based on notificaiton name and then post
~~~

An potential improvement to this would be to allow observations of all instances of a class that are contained within a specific object. For example, a glyph could observe the contour class rather than each individual contour. An API like this may work:

~~~ python
observable=(glyph, glyph.contourClass)
~~~

This shouldn't be too hard to implement inside of the dispatcher.

- [ ] Implement dispatcher changes.
- [ ] Add properties for all sub-object classes to the parent objects. Ex: font.kerningClass.
- [ ] Backport this to master.

**Comment by typesupply** [comment]
Any thoughts about this @typemytype, @letterror, @adrientetar?

**Comment by LettError** [comment]
This would work for all glyphs in a specific font, not all glyphs that happen to be open? There might be more glyph objects of the same class, but that belong to different fonts?

**Comment by typesupply** [comment]
> This would work for all glyphs in a specific font, not all glyphs that happen to be open?

Right. The dispatcher is unique to each font so it would only see and report notifications from the font that it belongs to.

**Comment by adrientetar** [comment]
Okay. I take it this is mainly useful inside of defcon. Indeed, it should help avoid issues like the one I fixed in https://github.com/trufont/defcon/commit/66c493d47b15e42a3385e0745740c156f946cea0.

> Backport this to master.

Still making changes to that branch?

**Comment by typesupply** [comment]
> Okay. I take it this is mainly useful inside of defcon.

I can use it in a lot of my RoboFont extensions as well. I have things like metrics previews that have to constantly subscribe/unsubscribe and the code for that is kind of gnarly. With this new thing I'd be able to have it generally observe font.glyphClass. No more "Is this glyph going away? Have I already subscribed to it? Argh notifications are awesome but this is too much code!"

> Still making changes to that branch?

Only big things that would be useful and can be backported easily. Also, probably only if someone else does it and gives me a PR. There has to be a name for this? Lazy Backporting?

**Comment by adrientetar** [comment]
I see well, sounds good.
